src/api/index.js

```
@@ -0,0 +1,92 @@
+import mocks from './mocks';
+
+const defaultServerUrl =
+  process.env.REACT_APP_SERVER_URL || 'http://127.0.0.1:8080/api';
+
+const apiUrl = url => `${defaultServerUrl}${url}`;
+
+function checkStatus(resp) {
+  if (resp.status < 200 || resp.status >= 300) {
+    const error = new Error(resp.statusText);
+    error.response = resp;
+    throw error;
+  }
+  return resp;
+}
+
+function normalizeError(err) {
+  if (typeof err === 'object') {
```

`normalizeError` seems to process different kinds of errors, but some of them are not properly shown (it is only displayed "Internal error").
What if we use `switch (typeof err)`? and then:
- properly define the acceptable types,
- gracefully display the `default:` kinds, instead of hiding them.

---

This function tries to make an error for human. If it isn't possible "Internal error" is the best option.

---

What if it's passed a number, for example?
```js
Internal error.
Content received: 2
```
Would be more detailed.

---

why do we need it?

---

I think it improves the clarity &rarr; dev experience

---

for dev you have `console.log`. End user doesn't need to know details. I'm sure @ricardobaeta can explain it better. But there are many resources on the internet with best practices from UX for error handling. None of them says you should show a user a traceback or errors that don't make sense for the user.

---

I'm not proposing to show such errors to the user.

The `normalizeError` is used internally to extract error messages, and it should be up to the UI components, to display them or not, or how to so.
Anyhow, `normalizeError` is already returning ["internal/unnexpected" errors](https://github.com/src-d/code-annotation/pull/9/files#diff-284817f1b6aad47c0ac6d61bd72f474cR27) as I'm talking about, and I think it's ok (but not done in a consistent way)